src/dispatching/dispatcher.rs

```diff
@@ -205,13 +225,13 @@ where
             }
         }
 
-        // TODO: wait for executing handlers?
-
+        self.workers.drain();
+        self.default_worker.take();
```

Comment:
This doesn't wait though.

I think the maps should not only store the `Sender<_>` (`WorkerTx`), but also the `JoinHandle` of the worker task.

Then we could

```rust
self.workers.drain()
    .chain(self.default_worker.take())
    .map(|(_tx, handle)| handle)
    .collect::<FuturesUnordered<_>>()
    .for_each(|()| ())
    .await
```

To wait for all the workers to stop.

Reply:
Fixed.